logger.js

```diff
@@ -64,16 +64,71 @@ function pinoLogger (opts, stream) {
   }
 }
 
+var rawSymbol = Symbol.for('pino-raw-request')
+var pinoReqProto = Object.create({}, {
+  id: {
+    enumerable: true,
+    writable: true,
+    value: ''
+  },
+  method: {
+    enumerable: true,
+    writable: true,
+    value: ''
+  },
+  url: {
+    enumerable: true,
+    writable: true,
+    value: ''
+  },
+  headers: {
+    enumerable: true,
+    writable: true,
+    value: {}
+  },
+  remoteAddress: {
+    enumerable: true,
+    writable: true,
+    value: ''
+  },
+  remotePort: {
+    enumerable: true,
+    writable: true,
+    value: ''
+  },
+  raw: {
```

Comment:
at this point I fail to see the point of the symbol

@mcollina  - thoughts?

Reply:
It gives a free non-enumerable. While writing this I learned:

```js
const proto = Object.create({}, {
  foo: {
    enumerable: false,
    writable: true,
    value: {}
  }
})

const instance = Object.create(proto)
console.log('%j', instance) // {}

instance.foo = 'foo'
console.log('%j', instance) // {"foo":"foo"}
```

Reply:
no sure – but you'd get that without the symbol and the getter/setter

Reply:
whats req.raw for btw... I saw it in the hapi-pino implementation as well - whats the reasoning

Reply:
ah no no wait, you'd need a getter - but you could just return req.raw

Reply:
No, same thing happens with the getter and setter unless you wrap the object in a function with an enclosed variable to contain the value instead of `set (val) { this._foo = val }`.

As for req.raw, see comments https://github.com/pinojs/hapi-pino/pull/34#issuecomment-347106887 and https://github.com/pinojs/hapi-pino/pull/34#issuecomment-347123516

Reply:
See https://gist.github.com/jsumners/0a586bf6c3dcb835d692251347a0e730

Reply:
gotcha!